### PR TITLE
Lint shell scripts with ShellCheck

### PR DESCRIPTION
**Pull request by phansys**
It seems that ShellCheck was already used in this project:
https://github.com/doctrine/coding-standard/blob/94bc305fb858ea58e6aa8829cff6c9677779203a/bin/test-report#L10

BTW, I'd like to discuss for a future PR if we should provide this tool and GNU make through the built-in Docker image (as this approach may ensure all the required tools are also available in the local environment).
https://github.com/doctrine/coding-standard/blob/94bc305fb858ea58e6aa8829cff6c9677779203a/.docker/validate-against-schema/Dockerfile#L1

### CODE REVIEW

**Review by ostrolucky — DISMISSED** [review]

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -82,6 +82,21 @@ jobs:
...
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v3"
+
+      - name: "Lint shell scripts with ShellCheck"
+        uses: "ludeeus/action-shellcheck@2.0.0"
+        with:
+          scandir: "./bin"
+          check_together: "yes"
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml]: Please restrict this action to merge request with changes to relevant files.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -82,6 +82,21 @@ jobs:
...
+    name: "Lint shell scripts"
+
+    runs-on: "ubuntu-22.04"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v3"
+
+      - name: "Lint shell scripts with ShellCheck"
+        uses: "ludeeus/action-shellcheck@2.0.0"
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml]: Do we need to specify the version number up to the patch version?

Comment (phansys) [reply .github/workflows/continuous-integration.yml]: I think it is required as there are no aliases ([branch](https://github.com/ludeeus/action-shellcheck/branches) or [tag](https://github.com/ludeeus/action-shellcheck/tags)) for `2.0` in that repo.

File: .github/workflows/ci-shell-scripts.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "*.*.x"
+    paths:
+      - "bin/**"
```
Comment (greg0ire) [inline .github/workflows/ci-shell-scripts.yml:15]: This workflow should run when modified.

```suggestion
      - ".github/workflows/ci-shell-scripts.yml"
      - "bin/**"
```

File: .github/workflows/ci-shell-scripts.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - ".github/workflows/ci-shell-scripts.yml"
+      - "bin/**"
+  push:
+    branches:
+      - "*.*.x"
+    paths:
+      - ".github/workflows/ci-shell-scripts.yml"
+      - "bin/**"
+  schedule:
+    - cron: "0 17 * * *"
```
Comment (greg0ire) [inline .github/workflows/ci-shell-scripts.yml:17]: :thinking: why?

Comment (phansys) [reply .github/workflows/ci-shell-scripts.yml:17]: The "general" CI pipeline is executed with the same schedule. IMO, is useful to have this pipeline on the same basis, as the rules or something else at ShellCheck or the GH action could be updated without a previous notice.

Comment (greg0ire) [reply .github/workflows/ci-shell-scripts.yml:17]: Oh right indeed… TIL!

**Review by greg0ire — APPROVED** [review]

**Review by ostrolucky — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @phansys !

**Review by derrabus — APPROVED** [review]

**Pull request merged**